modules/assim.sequential/R/Remote_helpers.R

```diff
@@ -0,0 +1,440 @@
+  
+  
+  
+  return(list(obs.mean = obs.mean,
+              obs.cov = obs.cov))
+}
+
+
+
+#' SDA_remote_launcher
+#'
+#' @param settingPath The Path to the setting that will run SDA
+#' @param ObsPath  Path to the obs data which is expected to be an .Rdata.
+#'
+#' @export
+#' @return This function returns a list of two pieces of information. One the remote path that SDA is running and the PID of the active run.
+#' @example 
+#' \dontrun{
+#'  # This example can be found under inst folder in the package
+#'  library(PEcAn.all)
+#'  library(purrr)
+#'  
+#'  run.bash.args <- c(
+#'   "#$ -l h_rt=48:00:00",
+#'   "#$ -pe omp 28 # Request a parallel environment with 4 cores",
+#'   "#$ -l mem_per_core=1G # and 4G memory for each",
+#'   "#$ -l buyin",
+#'   "module load R/3.5.2",
+#'   "module load python/2.7.13"
+#'   )
+#'  settingPath <-"pecan.SDA.4sites.xml"
+#'
+#'  ObsPath <- "Obs/LandTrendr_AGB_output50s.RData"
+#'    
+#'  SDA_remote_launcher(settingPath, ObsPath, run.bash.args)
+#'}
+#'
+SDA_remote_launcher <-function(settingPath, 
+                               ObsPath,
+                               run.bash.args){
+
+  #---------------------------------------------------------------
+  # Reading the settings
+  #---------------------------------------------------------------
+  settings <- read.settings(settingPath)
+  my_host <- list(name =settings$host$name , tunnel = settings$host$tunnel, user=settings$host$user)
+  local_path <-settings$outdir
+  if (is.null(run.bash.args)) run.bash.args <-""
+  #---------------------------------------------------------------
+  # Checking the setting xml
+  #---------------------------------------------------------------
+  if (is.null(settings$host$folder)) {
+    PEcAn.logger::logger.severe("You need to specify the <folder> tag in the <host> tag inside your pecan xml !")
+    PEcAn.logger::logger.severe("The <folder> tag is a path which points to where you want to store/run your sda job on the remote machine. ")
+  } else if (!test_remote(my_host)) {
+    PEcAn.logger::logger.severe("There is something wrong with your tunnel !")
+    PEcAn.logger::logger.severe("You can learn more about how to setup your tunnel by checking out the `Remote execution with PEcAn` section in the documentation.")
+    
+  } 
+  #----------------------------------------------------------------
+  # SDA xml tag
+  #---------------------------------------------------------------
+  if (is.null(settings$state.data.assimilation)) {
+    PEcAn.logger::logger.severe("Make sure that you have the state data assimilation tag in your xml ! You can learn more about it in the documantation.")
+  }
+  
+  #---------------------------------------------------------------
+  # Creating a new folder
+  #---------------------------------------------------------------
+
+  fname_p1 <- basename(settings$outdir)
+
+  
+  if (!is.null( settings$workflow$id)) {
+    fname_p2<-settings$workflow$id
+  } else {
+    fname_p2<-""
```

Comment:
```suggestion
    fname_p2 <- ""
```

but again could this be NULL?  empty is fine, just less explicit